scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala

```diff
@@ -96,7 +92,8 @@ object Term {
     checkFields(enums.head.is[Enumerator.Generator])
   }
   @ast class ForYield(enums: List[Enumerator] @nonEmpty, body: Term) extends Term
-  @ast class New(templ: Template) extends Term
+  @ast class New(init: Init) extends Term
+  @ast class NewAnonymous(templ: Template) extends Term
```

Comment:
I'd be in favor of `NewAnonymous`, since [SLS 6.10](https://www.scala-lang.org/files/archive/spec/2.11/06-expressions.html#instance-creation-expressions) says that this is instance creation of an anonymous class. 

I don't think that [SLS 3.2.7](https://www.scala-lang.org/files/archive/spec/2.11/03-types.html#compound-types) is applicable here, because refinements are a type-level concept, and `new t` from SLS 6.10 is operating not on types, but on templates.